### PR TITLE
Initialise genesis delegation from the first block

### LINKED ISSUE #504: Initial delegation certificates are applied incorrectly
## Background

In `Cardano.Chain.Delegation.Validation.Interface`, initial delegation is specified by:
- Firstly, constructing an initial delegation map which delegates from stakeholders to themselves.
- Secondly, applying the certificates from the genesis data in the normal way.

The problem is that normally, delegation certificates do not come into effect from the moment they are posted - they are only scheduled, and will come into effect at some future slot. For the genesis configuration, this is incorrect - the initial delegation should be in effect from the first block.

## Proposed Solution

Change `initialState` to create the delegation map directly using the heavyweight delegation certificates, rather than calling `updateDelegation`.

## Next Action
- [x] Confirm this diagnosis (@ruhatch?)
- [ ] Fix all the things

### PR DESCRIPTION
**Pull request by nc6**
I can confirm this addresses the issue we were seeing in ouroboros-consensus

Fixes #504

### CODE REVIEW

**Review by ruhatch — APPROVED** [review]
Love it, haha

File: src/Cardano/Chain/Delegation/Validation/Interface.hs [hunk 1]
```diff
@@ -71,9 +71,13 @@ initialState
   => Environment
   -> GenesisDelegation
   -> m State
-initialState env genesisDelegation = updateDelegation env is certificates
+initialState env genesisDelegation = updateDelegation env' is certificates
  where
   Environment { allowedDelegators } = env
+  -- We modify the environment here to allow the delegation certificates to
```
Comment (dnadales) [inline src/Cardano/Chain/Delegation/Validation/Interface.hs:77]: :heart:

**Review by dnadales — APPROVED** [review]

**Pull request merged**